crates/ruma-client-api/src/profile/get_profile_field.rs

```diff
@@ -0,0 +1,339 @@
+//! `GET /_matrix/client/*/profile/{userId}/{key_name}`
+//!
+//! Get a field in the profile of the user.
+
+pub mod v3 {
+    //! `/v3/` ([spec])
+    //!
+    //! [spec]: https://github.com/matrix-org/matrix-spec-proposals/pull/4133
+
+    use std::marker::PhantomData;
+
+    use ruma_common::{api::Metadata, metadata, OwnedUserId};
+
+    use crate::profile::{
+        AnyProfileField, ProfileField, StaticProfileField, StaticProfileFieldVisitor,
+    };
+
+    const METADATA: Metadata = metadata! {
+        method: GET,
+        rate_limited: false,
+        authentication: None,
+        history: {
+            unstable("uk.tcpip.msc4133") => "/_matrix/client/unstable/uk.tcpip.msc4133/profile/{user_id}/{field}",
+            // 1.15 => "/_matrix/client/v3/profile/{user_id}/{field}",
+        }
+    };
+
+    /// Request type for the `get_profile_field` endpoint.
+    #[derive(Debug, Clone)]
+    #[cfg_attr(not(ruma_unstable_exhaustive_types), non_exhaustive)]
+    pub struct Request<F: ProfileField> {
```

Comment:
So I chose to experiment using a trait on the request to be able to validate the response ourselves, instead of having an enum in the response and the caller needs to extract the proper variant for the field that they have requested.

We don't do this anywhere else (but we could, like when we request a particular state event). Maybe it complicates the API a bit too much?